pkg/engine/lifecycletest/target_test.go

```diff
@@ -112,12 +112,13 @@ func TestRefreshTargetChildren(t *testing.T) {
 	require.Len(t, snap.Resources, 5)
 	assert.Equal(t, snap.Resources[1].URN, a)
 	assert.Equal(t, snap.Resources[1].Outputs["count"], null)
+
 	assert.Equal(t, snap.Resources[2].URN, b)
-	assert.Equal(t, snap.Resources[2].Outputs["count"], resource.NewPropertyValue(1.0))
+	assert.NotEqual(t, snap.Resources[2].Outputs["count"], null)
 	assert.Equal(t, snap.Resources[3].URN, c)
-	assert.Equal(t, snap.Resources[3].Outputs["count"], resource.NewPropertyValue(2.0))
+	assert.NotEqual(t, snap.Resources[3].Outputs["count"], null)
 	assert.Equal(t, snap.Resources[4].URN, d)
-	assert.Equal(t, snap.Resources[4].Outputs["count"], resource.NewPropertyValue(3.0))
+	assert.NotEqual(t, snap.Resources[4].Outputs["count"], null)
```

Comment:
Instead of setting the `count` output property at all, should we assert that `callCount == 3` here?  The intention of the `count` output seems to have been to make sure that we call `ReadF` a certain number of times?

Reply:
Perhaps also have `ReadF` return a value that includes the URN with some deterministic thing? Then you can assert that each thing was read with a specific response?

Reply:
Ah, that's a neat idea! Updated the PR

Reply:
@lunaris Another neat idea, though I might leave that one for a follow-up in the interests of fixing the flake ASAP